app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitListFragment.java

```diff
@@ -144,15 +182,23 @@ public void addNewHabit(Habit habit) {
      */
     public void addHabitToDatabase(String title, String reason, Date date){
         // Handling of adding a habit to firebase
-        final CollectionReference collectionReference = _db.collection("Habits");
         HashMap<String, Object> habitData = new HashMap<>();
 
+        // find lowest positive non zero habitId that is not used by a habit currently
+        long habitIdCounter = 1;
+        while (_habitIdSet.contains(habitIdCounter)) {
```

Comment:
I think we should avoid `while` loops when possible. I'd like this to be changed to a `for` loop if we know how many elements we are iterating on. Even though it's not a problem now, there shouldn't be a possibility of an infinite loop in the project.

Reply:
Alternatively, firebase can set an automatically generated `id`, have you looked into that?